### PR TITLE
Release 4.10

### PR DESCRIPTION
**Pull request by daosman**
Created a 4.10-release branch to create a Pull Request so that the code can be reviewed and commented on.

### CODE REVIEW

File: inventory.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    pull_secret: /tmp/pull-secret.txt
+    
+    # Path to kubeconfig for the ACM hub cluster
+    kubeconfig: /tmp/kubeconfig
+    
+    # Path to repo git pull directory
+    repopath: /tmp/openshift-acm-oot-drv
+
+    # Local registry domain name and port.
+    registry: "cnfde2.ptp.lab.eng.bos.redhat.com:5000"
```
Comment (aneeshkp) [inline inventory.yml]: Add details to the readme to show what variables need to change/update.  for example, the local registry domain name and port should be updated.

File: roles/undeploy_sro/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+sro_channel: "4.9"
```
Comment (aneeshkp) [inline roles/undeploy_sro/defaults/main.yml]: Does the channel need to be variable? The operator channel changes for OCP version

File: README.md [hunk 3]
```diff
@@ -1 +1,124 @@
-# openshift-acm-oot-drv
\ No newline at end of file
+# openshift-acm-oot-drv
+
+This repository contains playbooks to manage out of tree drivers using the Special Resource Operator on Redhat ACM managed clusters.
```
Comment (leifmadsen) [inline README.md]:
```suggestion
This repository contains playbooks to manage out-of-tree drivers using the Special Resource Operator on Red Hat ACM (RHACM) managed clusters.
```

File: roles/undeploy_sro/defaults/main.yml (hunk 2 above)
Comment (daosman) [reply roles/undeploy_sro/defaults/main.yml]: This is a variable.  how about if I update it to 4.10 for the 4.10 branch and so on ?

File: inventory.yml (hunk 1 above)
Comment (daosman) [reply inventory.yml]: Will do

File: README.md [hunk 4]
```diff
@@ -1 +1,124 @@
...
+
+The target deployment model is a RHACM hub cluster managing many spoke SNO clusters as illustrated below.
```
Comment (leifmadsen) [inline README.md]:
```suggestion
The target deployment model is a RHACM hub cluster managing multi-spoke Single Node OpenShift (SNO) clusters.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,124 @@
...
+                                                                     └┬──────────┘││
+                                                                      └───────────┘│
+                                                                       └───────────┘
+
+
+```
+
+The playbooks deploy SRO on the ACM Hub cluster and builds the drivers there.  Then uses a MachineConfig wrapped in an ACM policy to deploy the driver.
+
+It currently only supports Single Node Openshift (SNO) nodes as managed clusters.
```
Comment (leifmadsen) [inline README.md]:
```suggestion
The playbooks only supports SNO nodes as managed clusters.
```

File: README.md [hunk 6]
```diff
@@ -1 +1,124 @@
...
+pip install kubernetes
+```
+
+There's also some required Ansible modules that can be installed with the following command:
+
+```bash
+ansible-galaxy collection install -r requirements.yml
+```
+
+## Before Running The Playbook
```
Comment (leifmadsen) [inline README.md]:
```suggestion
## Prerequisites
```

File: README.md [hunk 7]
```diff
@@ -1 +1,124 @@
...
+
+There are few setup steps required prior to running the playbook with the sample driver.
```
Comment (leifmadsen) [inline README.md]:
```suggestion
The following steps are required prior to running the playbook.
```

File: README.md [hunk 8]
```diff
@@ -1 +1,124 @@
...
+
+- Make sure you can run the OpenShift CLI as a user with cluster-admin privileges
```
Comment (leifmadsen) [inline README.md:65]:
```suggestion
- Verify the `oc` client can be run as a user with cluster-admin privileges
```

File: README.md [hunk 9]
```diff
@@ -1 +1,124 @@
...
+- Setup the necessary prerequisites by running the `deploy_prerequisites.yml` playbook
```
Comment (leifmadsen) [inline README.md:66]:
```suggestion
- Run the `deploy_prerequisites.yml` playbook
```

File: README.md [hunk 10]
```diff
@@ -1 +1,124 @@
...
+export REGISTRY_PORT='5000'
+export REGISTRY_CERT='/path/to/your/registry/domain.crt'
+
+oc create configmap registry-cas -n openshift-config --from-file=${REGISTRY_FQDN}..${REGISTRY_PORT}=${REGISTRY_CERT}
+oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-cas"}}}' --type=merge
+```
+
+- Add the `specialresource.openshift.io/intel-eth-800` label that the sample driver is expecting onto the nodes that have the Intel(R) Ethernet 800 Series NIC
+
+## Running The Playbooks
```
Comment (leifmadsen) [inline README.md:92]:
```suggestion
## Run the playbooks

File: README.md [hunk 11]
```diff
@@ -1 +1,124 @@
...
+
+To run the playbooks with the provided sample driver, use these commands.
+
+```bash
+ansible-playbook -i inventory deploy_sro.yml
+ansible-playbook -i inventory build_drivers.yml
+ansible-playbook -i inventory deploy_drivers.yml
+```
+
+## Adding your own drivers
```
Comment (leifmadsen) [inline README.md:108]:
```suggestion
## Add the drivers
```

File: README.md [hunk 12]
```diff
@@ -1 +1,124 @@
...
+
+To add you own drivers, you can follow the sample driver provided.
```
Comment (leifmadsen) [inline README.md:110]:
```suggestion
To add your own drivers, you can follow the sample driver provided.
```

File: roles/build_driver/README.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+This role will generate a SpecialResouce CR from the template given by the
+user.  It then applies the CR onto the cluster to kick start the driver builds
+Once the get_managed_clusters builds a list of OCP version from the
+ManagedClusters CR, it builds a unique list of kernel version as some different
+OCP version might have the same underlying kernel version.
+
+## Requirements
+
+SRO and NFD have to be installed and properly functioning.
```
Comment (leifmadsen) [inline roles/build_driver/README.md]: Define `NFD` first time it is used.

File: roles/deploy_drivers/README.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+MachineConfig template given by the user.
+
+## Requirements
+
+The driver containers are built and pushed to the external registry.
+
+## Role variables
+
+A state variable is passed to either add or remove all the objects that this
+role uses. A 'absent' state will remove all the objects while a 'present'
```
Comment (leifmadsen) [inline roles/deploy_drivers/README.md]:
```suggestion
role uses. An 'absent' state will remove all the objects while a 'present'
```

File: README.md [hunk 15]
```diff
@@ -1 +1,124 @@
...
+                                                               └─────┤ SNO SET-C │├┐
+                                                                     │           │││
+                                                                     └┬──────────┘││
+                                                                      └───────────┘│
+                                                                       └───────────┘
+
+
+```
+
+The playbooks deploy SRO on the ACM Hub cluster and builds the drivers there.  Then uses a MachineConfig wrapped in an ACM policy to deploy the driver.
```
Comment (leifmadsen) [inline README.md]:
```suggestion
Ansible playbooks deploy SpecialResourceOperator (SRO) on the ACM Hub cluster and builds the drivers there, then uses a MachineConfig wrapped in an ACM policy to deploy the driver.
```

File: roles/build_driver/README.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
+## build_driver
+
+This role will generate a SpecialResouce CR from the template given by the
```
Comment (leifmadsen) [inline roles/build_driver/README.md]:
```suggestion
This role will generate a SpecialResource CR from the template given by the
```

File: roles/build_driver/README.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+user.  It then applies the CR onto the cluster to kick start the driver builds
```
Comment (leifmadsen) [inline roles/build_driver/README.md]:
```suggestion
user.  It then applies the CR onto the cluster to kick start the driver builds.
```

**Pull request merged**